# Immunophenotypic correlates of sustained MRD negativity in patients with multiple myeloma

------------------------------------------------------------------------

#### David G. Coffey<sup>1,2</sup>, Francesco Maura<sup>1</sup>, Edgar E. Gonzalez-Kozlova<sup>3</sup>, J. Javier Diaz-Mejia<sup>4</sup>, Ping Luo<sup>4</sup>, Yong Zhang<sup>5</sup>, Yuexin Xu<sup>2</sup>, Edus H. Warren<sup>2</sup>, Travis Dawson<sup>3</sup>, Brian Lee<sup>3</sup>, Hui Xie<sup>3</sup>, Eric Smith<sup>6</sup>, Amanda Ciardiello<sup>7</sup>, Hearn J. Cho<sup>3</sup>, Adeeb Rahman<sup>3</sup>, Seunghee Kim-Schulze<sup>3</sup>, Benjamin Diamond<sup>1</sup>, Alexander Lesokhin<sup>7†</sup>, Dickran Kazandjian<sup>1†</sup>, Trevor Pugh<sup>4†</sup>, Damian J. Green<sup>2†</sup>, Sacha Gnjatic<sup>3†</sup>, Ola Landgren<sup>1†</sup>

###### <sup>1</sup>Division of Myeloma, Sylvester Comprehensive Cancer Center, University of Miami, Miami, FL; <sup>2</sup>Clinical Research Division, Fred Hutchinson Cancer Research Center, Seattle, WA; <sup>3</sup>Icahn School of Medicine at Mount Sinai, New York, NY; <sup>4</sup>Princess Margaret Cancer Centre, University of Toronto, Toronto, Canada; <sup>5</sup>Center for Drug Evaluation and Research, US Food and Drug Administration, Silver Spring, MD; <sup>6</sup>Dana-Farber Cancer Institute, Boston, MA; <sup>7</sup>Myeloma Service, Department of Medicine, Memorial Sloan Kettering Cancer Center, New York, NY; <sup>†</sup>These authors contributed equally

------------------------------------------------------------------------

#### Supplemental code

This repository contains R code for analyzing single-cell RNA sequencing (scRNAseq) combined with V(D)J sequencing, CyTOF mass cytometry, and T cell receptor $\beta$ sequencing.

-   `scRNAseq Methods.R` This file demonstrates how gene expression matrix files output from the [cell ranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) v4.0.0 (10X Genomics) pipeline were processed into cell count matrices for differential cellular abundance analysis. Also shown is code for differential gene expression per cell type and methods for identifying exhausted T cells.

-   `scRNAseq VDJ Methods.R` This file demonstrates how all_contig_annotations.csv V(D)J files generated by [cell ranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) v4.0.0 (10X Genomics) pipeline were filtered to exclude doublets, unpaired, and multi-chain T and B cell receptors.

-   `CyTOF Methods.R` This file demonstrates how output from the [Astrolabe Mass Cytometry Platform](https://astrolabediagnostics.com) (Astrolabe Diagnostics, Inc.) was input into R using the [orloj](https://github.com/astrolabediagnostics/orloj) package, filtered, and analyzed for differential cellular abundance.

-   `TCRB Methods.R` This file demonstrates how tab delimited files output from the [ImmunoSeq](https://www.immunoseq.com) platform (Adaptive Biotechnology) were input into R using the [LymphoSeq](http://www.bioconductor.org/packages/release/bioc/html/LymphoSeq.html) package, repertoire statistics were calculated, and differential abundance analysis was performed.
