# Immunophenotypic correlates of sustained MRD negativity in patients with multiple myeloma

------------------------------------------------------------------------

#### David G. Coffey^1,2^, Francesco Maura^1^, Edgar E. Gonzalez-Kozlova^3^, J. Javier Diaz-Mejia^4^, Ping Luo^4^, Yong Zhang^5^, Yuexin Xu^2^, Edus H. Warren^2^, Travis Dawson^3^, Brian Lee^3^, Hui Xie^3^, Eric Smith^6^, Amanda Ciardiello^7^, Hearn J. Cho^3^, Adeeb Rahman^3^, Seunghee Kim-Schulze^3^, Benjamin Diamond^1^, Alexander Lesokhin^7†^, Dickran Kazandjian^1†^, Trevor Pugh^4†^, Damian J. Green^2†^, Sacha Gnjatic^3†^, Ola Landgren^1†^

###### ^1^Division of Myeloma, Sylvester Comprehensive Cancer Center, University of Miami, Miami, FL; ^2^Clinical Research Division, Fred Hutchinson Cancer Research Center, Seattle, WA; ^3^Icahn School of Medicine at Mount Sinai, New York, NY; ^4^Princess Margaret Cancer Centre, University of Toronto, Toronto, Canada; ^5^Center for Drug Evaluation and Research, US Food and Drug Administration, Silver Spring, MD; ^6^Dana-Farber Cancer Institute, Boston, MA; ^7^Myeloma Service, Department of Medicine, Memorial Sloan Kettering Cancer Center, New York, NY; ^†^These authors contributed equally

------------------------------------------------------------------------

#### Supplemental code

This repository contains R code for analyzing single-cell RNA sequencing (scRNAseq) combined with V(D)J sequencing, CyTOF mass cytometry, and T cell receptor $\beta$ sequencing.

-   `scRNAseq Methods.R` This file demonstrates how gene expression matrix files output from the [cell ranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) v4.0.0 (10X Genomics) pipeline were processed into cell count matrices for differential cellular abundance analysis. Also shown is code for differential gene expression per cell type and methods for identifying exhausted T cells.

-   `scRNAseq VDJ Methods.R` This file demonstrates how all_contig_annotations.csv V(D)J files generated by [cell ranger](https://support.10xgenomics.com/single-cell-gene-expression/software/pipelines/latest/what-is-cell-ranger) v4.0.0 (10X Genomics) pipeline were filtered to exclude doublets, unpaired, and multi-chain T and B cell receptors.

-   `CyTOF Methods.R` This file demonstrates how output from the [Astrolabe Mass Cytometry Platform](https://astrolabediagnostics.com) (Astrolabe Diagnostics, Inc.) was input into R using the [orloj](https://github.com/astrolabediagnostics/orloj) package, filtered, and analyzed for differential cellular abundance.

-   `TCRB Methods.R` This file demonstrates how tab delimited files output from the [ImmunoSeq](https://www.immunoseq.com) platform (Adaptive Biotechnology) were input into R using the [LymphoSeq](http://www.bioconductor.org/packages/release/bioc/html/LymphoSeq.html) package, repertoire statistics were calculated, and differential abundance analysis was performed.
